File: openapi-test-java-server-quarkus-reactive-client-mp-reactive/testapi/api/PetsApiClient.java
package testapi.api;

import com.fasterxml.jackson.databind.JsonNode;
import io.smallrye.mutiny.Uni;
import io.swagger.v3.oas.annotations.security.SecurityRequirement;
import jakarta.ws.rs.Consumes;
import jakarta.ws.rs.DELETE;
import jakarta.ws.rs.DefaultValue;
import jakarta.ws.rs.GET;
import jakarta.ws.rs.POST;
import jakarta.ws.rs.Path;
import jakarta.ws.rs.PathParam;
import jakarta.ws.rs.Produces;
import jakarta.ws.rs.QueryParam;
import jakarta.ws.rs.WebApplicationException;
import jakarta.ws.rs.core.MediaType;
import jakarta.ws.rs.core.Response;
import java.lang.IllegalStateException;
import java.lang.Void;
import java.util.List;
import java.util.Optional;
import java.util.function.Function;
import org.eclipse.microprofile.rest.client.inject.RegisterRestClient;
import org.glassfish.jersey.media.multipart.FormDataParam;
import testapi.model.Error;
import testapi.model.Pet;
import testapi.model.PetCreate;

@RegisterRestClient
@Path("/pets")
public interface PetsApiClient extends PetsApi {
  /** Create a pet */
  @POST
  @Path("/")
  @Consumes(MediaType.APPLICATION_JSON)
  @Produces(MediaType.APPLICATION_JSON)
  @SecurityRequirement(name = "oauth2", scopes = { "write:pets" })
  @SecurityRequirement(name = "apiKeyHeader")
  Uni<Response> createPetRaw(PetCreate body);

  /** Create a pet - handles response status codes */
  @Override
  default Uni<Response201400<Pet, Error>> createPet(PetCreate body) {
    return createPetRaw(body).onFailure(WebApplicationException.class).recoverWithItem((Throwable e) -> ((WebApplicationException) e).getResponse()).map((Response response) -> {
      if (response.getStatus() == 201) { return new Created(response.readEntity(Pet.class)); }
    else if (response.getStatus() == 400) { return new BadRequest(response.readEntity(Error.class)); }
    else { throw new IllegalStateException("Unexpected status code: " + response.getStatus()); }
    });
  };

  /** Delete a pet */
  @DELETE
  @Path("/{petId}")
  Uni<Response> deletePetRaw(
  
    /** The pet ID */
    @PathParam("petId") String petId
  );

  /** Delete a pet - handles response status codes */
  @Override
  default Uni<Response404Default<Error>> deletePet(
  
    /** The pet ID */
    String petId
  ) {
    return deletePetRaw(petId).onFailure(WebApplicationException.class).recoverWithItem((Throwable e) -> ((WebApplicationException) e).getResponse()).map((Response response) -> {
      if (response.getStatus() == 404) { return new NotFound(response.readEntity(Error.class)); }
    else { return new Default(response.getStatus(), response.readEntity(Error.class)); }
    });
  };

  /** Get a pet by ID */
  @GET
  @Path("/{petId}")
  @Produces(MediaType.APPLICATION_JSON)
  Uni<Response> getPetRaw(
  
    /** The pet ID */
    @PathParam("petId") String petId
  );

  /** Get a pet by ID - handles response status codes */
  @Override
  default Uni<Response200404<Pet, Error>> getPet(
  
    /** The pet ID */
    String petId
  ) {
    return getPetRaw(petId).onFailure(WebApplicationException.class).recoverWithItem((Throwable e) -> ((WebApplicationException) e).getResponse()).map((Response response) -> {
      if (response.getStatus() == 200) { return new Ok(response.readEntity(Pet.class)); }
    else if (response.getStatus() == 404) { return new NotFound(response.readEntity(Error.class)); }
    else { throw new IllegalStateException("Unexpected status code: " + response.getStatus()); }
    });
  };

  /** Get pet photo */
  @Override
  @GET
  @Path("/{petId}/photo")
  @Produces(MediaType.APPLICATION_OCTET_STREAM)
  Uni<Void> getPetPhoto(
  
    /** The pet ID */
    @PathParam("petId") String petId
  );

  /** List all pets */
  @Override
  @GET
  @Path("/")
  @Produces(MediaType.APPLICATION_JSON)
  Uni<List<Pet>> listPets(
    /** Maximum number of pets to return */
    @QueryParam("limit") @DefaultValue("20") Optional<Integer> limit,
    /** Filter by status */
    @QueryParam("status") @DefaultValue("available") Optional<String> status
  );

  /** Upload a pet photo */
  @Override
  @POST
  @Path("/{petId}/photo")
  @Consumes(MediaType.MULTIPART_FORM_DATA)
  @Produces(MediaType.APPLICATION_JSON)
  Uni<JsonNode> uploadPetPhoto(
    /** The pet ID */
    @PathParam("petId") String petId,
    /** Optional caption for the photo */
    @FormDataParam("caption") String caption,
    /** The photo file to upload */
    @FormDataParam("file") Byte[] file
  );
}

File: openapi-test-kotlin-server-quarkus-reactive-client-mp-reactive/testapi/api/PetsApiClient.kt
package testapi.api

import com.fasterxml.jackson.databind.JsonNode
import io.smallrye.mutiny.Uni
import io.swagger.v3.oas.annotations.security.SecurityRequirement
import jakarta.ws.rs.Consumes
import jakarta.ws.rs.DELETE
import jakarta.ws.rs.GET
import jakarta.ws.rs.POST
import jakarta.ws.rs.Path
import jakarta.ws.rs.Produces
import jakarta.ws.rs.WebApplicationException
import jakarta.ws.rs.core.MediaType
import jakarta.ws.rs.core.Response
import java.lang.IllegalStateException
import java.lang.Void
import java.util.Optional
import java.util.function.Function
import kotlin.collections.List
import testapi.model.Error
import testapi.model.Pet
import testapi.model.PetCreate

interface PetsApiClient : PetsApi {
  /** Create a pet - handles response status codes */
  override fun createPet(body: PetCreate): Uni<Response201400<Pet, Error>> = createPetRaw(body).onFailure(WebApplicationException::class.java).recoverWithItem(object : Function<Throwable, Response> { override fun apply(e: Throwable): Response = (e as WebApplicationException).getResponse() }).map({ response: Response -> if (response.getStatus() == 201) { Created(response.readEntity(Pet::class.java)) }
  else if (response.getStatus() == 400) { BadRequest(response.readEntity(Error::class.java)) }
  else { throw IllegalStateException("Unexpected status code: " + response.getStatus()) } })

  /** Create a pet */
  @POST
  @Path("/")
  @Consumes(MediaType.APPLICATION_JSON)
  @Produces(MediaType.APPLICATION_JSON)
  @SecurityRequirement(name = "oauth2", scopes = ["write:pets"])
  @SecurityRequirement(name = "apiKeyHeader")
  fun createPetRaw(body: PetCreate): Uni<Response>

  /** Delete a pet - handles response status codes */
  override fun deletePet(
    /** The pet ID */
    petId: String
  ): Uni<Response404Default<Error>> = deletePetRaw(petId).onFailure(WebApplicationException::class.java).recoverWithItem(object : Function<Throwable, Response> { override fun apply(e: Throwable): Response = (e as WebApplicationException).getResponse() }).map({ response: Response -> if (response.getStatus() == 404) { NotFound(response.readEntity(Error::class.java)) }
  else { Default(response.getStatus(), response.readEntity(Error::class.java)) } })

  /** Delete a pet */
  @DELETE
  @Path("/{petId}")
  fun deletePetRaw(
    /** The pet ID */
    petId: String
  ): Uni<Response>

  /** Get a pet by ID - handles response status codes */
  override fun getPet(
    /** The pet ID */
    petId: String
  ): Uni<Response200404<Pet, Error>> = getPetRaw(petId).onFailure(WebApplicationException::class.java).recoverWithItem(object : Function<Throwable, Response> { override fun apply(e: Throwable): Response = (e as WebApplicationException).getResponse() }).map({ response: Response -> if (response.getStatus() == 200) { Ok(response.readEntity(Pet::class.java)) }
  else if (response.getStatus() == 404) { NotFound(response.readEntity(Error::class.java)) }
  else { throw IllegalStateException("Unexpected status code: " + response.getStatus()) } })

  /** Get pet photo */
  @GET
  @Path("/{petId}/photo")
  @Produces(MediaType.APPLICATION_OCTET_STREAM)
  override fun getPetPhoto(
    /** The pet ID */
    petId: String
  ): Uni<Void>

  /** Get a pet by ID */
  @GET
  @Path("/{petId}")
  @Produces(MediaType.APPLICATION_JSON)
  fun getPetRaw(
    /** The pet ID */
    petId: String
  ): Uni<Response>

  /** List all pets */
  @GET
  @Path("/")
  @Produces(MediaType.APPLICATION_JSON)
  override fun listPets(
    /** Maximum number of pets to return */
    limit: Optional<Integer>,
    /** Filter by status */
    status: Optional<String>
  ): Uni<List<Pet>>

  /** Upload a pet photo */
  @POST
  @Path("/{petId}/photo")
  @Consumes(MediaType.MULTIPART_FORM_DATA)
  @Produces(MediaType.APPLICATION_JSON)
  override fun uploadPetPhoto(
    /** The pet ID */
    petId: String,
    /** Optional caption for the photo */
    caption: String,
    /** The photo file to upload */
    file: Array<Byte>
  ): Uni<JsonNode>
}